### PR TITLE
[PKG-5222] 6.7.3 Initial build ❄️

### PR DESCRIPTION
**Pull request by ViridianMelody**
pyomo 6.7.3 :snowflake:

**Destination channel:** Snowflake

### Links

- [PKG-5222] 
- [Upstream repository](https://github.com/Pyomo/pyomo)
- [Upstream changelog/diff]()

### Explanation of changes:

- Added upstream tests
  - Skipping several tests on Windows that are failing on file comparisons.

[PKG-5222]: https://anaconda.atlassian.net/browse/PKG-5222?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by psteyer — APPROVED** [review]
LGTM

File: recipe/meta.yaml [hunk 1]
```diff
@@ -154,16 +153,48 @@ test:
     - pyomo.solvers.plugins.solvers
     - pyomo.util
     - pyomo.version
-
+  source_files:
+  # Tests are spread across many files throughout the source code.
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]: `pytest --pyargs pyomo` should be able to find all the tests.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,51 +1,46 @@
...
-  url: https://pypi.io/packages/source/P/Pyomo/Pyomo-{{ version }}.tar.gz
-  sha256: 5a23e775bba9fdbad22698fa1a841e662482edc979f2dea41cc6c54b1bb4b968
+source:
+  url: https://github.com/Pyomo/pyomo/archive/refs/tags/6.7.3.tar.gz
+  sha256: d31cdd1c63a80d39e51ff79b1e9dc10a1a0814fafe7f68777e682129e2179614
+
 
 build:
   number: 0
-  script: "{{ PYTHON }} -m pip install . --install-option=--with-distributable-extensions --no-deps -vv"
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml:16]: Why did you remove `--install-option=--with-distributable-extensions`? This flag enables https://github.com/Pyomo/pyomo/blob/6.7.3/setup.py#L110-L131.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -154,16 +153,48 @@ test:
...
     - pyomo.version
-
+  source_files:
+  # Tests are spread across many files throughout the source code.
+    - .
+  requires:
+    - pip
+    - pytest
+    - parameterized
+    - pybind11
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]: Why is pybind11 needed to run the tests?

File: recipe/meta.yaml [hunk 4]
```diff
@@ -154,16 +153,48 @@ test:
...
   commands:
     - pyomo --help
+    - pip check
+    {% set tests_to_skip = "abstract_ch_pyomo_buildactions_fails_sh" %}
+    {% set tests_to_skip = tests_to_skip + " or abstract_ch_pyomo_model" %}
+    # Missing file "lsb_release"
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]: §if we add `distro` as a dependency, it should work, see https://github.com/Pyomo/pyomo/blob/1ada528693d8879f17978fe47263f8bb2b467b0a/pyomo/common/download.py#L154-L155.

File: recipe/meta.yaml (hunk 2 above)
Comment (ViridianMelody) [reply recipe/meta.yaml:16]: I'm getting an error with it present: `no such option: --install-option`

File: recipe/meta.yaml (hunk 3 above)
Comment (ViridianMelody) [reply recipe/meta.yaml]: Unsure, I was just following the upstream testing dependencies here: https://github.com/Pyomo/pyomo/blob/1ada528693d8879f17978fe47263f8bb2b467b0a/setup.py#L249

I can give it a shot without pybind11.

File: recipe/meta.yaml (hunk 1 above)
Comment (ViridianMelody) [reply recipe/meta.yaml]: Is there a way to specify files to ignore during collection when using `--pyargs`? `test_piecewise_linear_kernel.py` triggers an error during collection otherwise:

```
_ ERROR collecting solvers/tests/piecewise_linear/test_piecewise_linear_kernel.py _
../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib/python3.8/site-packages/pyomo/solvers/tests/piecewise_linear/test_piecewise_linear_kernel.py:147: in <module>
    assignTests(PiecewiseLinearKernelTest, problems)
../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib/python3.8/site-packages/pyomo/solvers/tests/piecewise_linear/test_piecewise_linear_kernel.py:131: in assignTests
    with open(
E   FileNotFoundError: [Errno 2] No such file or directory: '$PREFIX/lib/python3.8/site-packages/pyomo/solvers/tests/piecewise_linear/kernel_baselines/convex_var_baseline_results.json'
```

File: recipe/meta.yaml (hunk 2 above)
Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml:16]: In these cases, we need to understand how that flag was used and how we can continue to use it.

In this specific case, you can pass it through an environment variable, see https://github.com/Pyomo/pyomo/blob/main/setup.py#L110-L113.

Comment (ViridianMelody) [reply recipe/meta.yaml:16]: Got it, thank you!

File: recipe/meta.yaml (hunk 1 above)
Comment (lorepirri) [reply recipe/meta.yaml]: @ViridianMelody have a look at this commit https://github.com/AnacondaRecipes/alibi-detect-feedstock/pull/1/commits/4e14153c892b00cac0dd57117da92586233826e7

File: recipe/meta.yaml (hunk 2 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml:16]: `'--global-option="--with-cython --with-distributable-extensions"'` should work too, see https://github.com/Pyomo/pyomo/blob/6.7.3/.github/workflows/release_wheel_creation.yml#L56

File: recipe/meta.yaml [hunk 5]
```diff
@@ -154,16 +154,47 @@ test:
...
+    {% set tests_to_skip = tests_to_skip + " or test_large_expression" %}  # [win]
+    {% set tests_to_skip = tests_to_skip + " or test_pickle4" %}  # [win]
+    {% set tests_to_skip = tests_to_skip + " or test_soln_pprint1" %}  # [win]
+    {% set tests_to_skip = tests_to_skip + " or test_write_solution" %}  # [win]
+
+    {% set tests_to_ignore = "--ignore=pyomo/solvers/tests/piecewise_linear/test_piecewise_linear_kernel.py" %}
+    {% set tests_to_ignore = tests_to_ignore + " --ignore=pyomo/environ" %}
+    {% set tests_to_ignore = tests_to_ignore + " --ignore=pyomo/repn/tests/mps/test_mps.py" %}  # [win]
+
+    - pytest -k "not ({{ tests_to_skip }})" {{ tests_to_ignore }}
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]: One test failed on `linux-64`:
`FAILED pyomo/common/tests/test_download.py::Test_FileDownloader::test_get_os_version`
